### ISSUE TITLE
My client is crashing because of Optifine 1.12.2

### ISSUE THREAD

**Issue by gpxorg** [sp614x/optifine#3539]
Ive been switching modfiles. Let's just say that I have 2 files with different mods. I switch them with the current ones and now my client isn't starting. Why's that? Heres the crash log

`[19:35:09] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLTweaker
[19:35:09] [main/INFO] [LaunchWrapper]: Using primary tweak class name net.minecraftforge.fml.common.launcher.FMLTweaker
[19:35:09] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLTweaker
[19:35:09] [main/INFO] [FML]: Forge Mod Loader version 14.23.5.2847 for Minecraft 1.12.2 loading
[19:35:09] [main/INFO] [FML]: Java is Java HotSpot(TM) 64-Bit Server VM, version 1.8.0_51, running on Windows 10:amd64:10.0, installed at C:\Program Files (x86)\Minecraft Launcher\runtime\jre-x64
[19:35:09] [main/WARN] [FML]: The coremod FMLCorePlugin (net.minecraftforge.fml.relauncher.FMLCorePlugin) is not signed!
[19:35:09] [main/WARN] [FML]: The coremod FMLForgePlugin (net.minecraftforge.classloading.FMLForgePlugin) is not signed!
[19:35:09] [main/INFO] [FML]: Searching C:\Users\User\AppData\Roaming\.minecraft\mods for mods
[19:35:09] [main/INFO] [FML]: Searching C:\Users\User\AppData\Roaming\.minecraft\mods\1.12.2 for mods
[19:35:09] [main/WARN] [FML]: Found FMLCorePluginContainsFMLMod marker in LabyMod-3.jar. This is not recommended, @Mods should be in a separate jar from the coremod.
[19:35:10] [main/WARN] [FML]: The coremod net.labymod.core.asm.LabyModCoreMod does not have a MCVersion annotation, it may cause issues with this version of Minecraft
[19:35:10] [main/WARN] [FML]: The coremod LabyModCoreMod (net.labymod.core.asm.LabyModCoreMod) is not signed!
[19:35:10] [main/INFO] [FML]: Loading tweaker optifine.OptiFineForgeTweaker from OptiFine 1.12.2 HD U E2.jar
[19:35:10] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[19:35:10] [main/INFO] [LaunchWrapper]: Loading tweak class name optifine.OptiFineForgeTweaker
[19:35:10] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLDeobfTweaker
[19:35:10] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[19:35:10] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[19:35:10] [main/INFO] [LaunchWrapper]: Calling tweak class optifine.OptiFineForgeTweaker
[19:35:10] [main/INFO] [net.labymod.support.util.CapturePrintStream]: OptiFineForgeTweaker: acceptOptions
[19:35:10] [main/INFO] [net.labymod.support.util.CapturePrintStream]: OptiFineForgeTweaker: injectIntoClassLoader
[19:35:10] [main/INFO] [net.labymod.support.util.CapturePrintStream]: OptiFine ClassTransformer
[19:35:10] [main/INFO] [net.labymod.support.util.CapturePrintStream]: OptiFine ZIP file: C:\Users\User\AppData\Roaming\.minecraft\mods\1.12.2\OptiFine 1.12.2 HD U E2.jar
[19:35:10] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[19:35:11] [main/INFO] [FML]: Found valid fingerprint for Minecraft. Certificate fingerprint cd99959656f753dc28d863b46769f7f8fbaefcfc
[19:35:11] [main/ERROR] [FML]: FML appears to be missing any signature data. This is not a good thing
[19:35:11] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[19:35:11] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[19:35:11] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLDeobfTweaker
[19:35:12] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.TerminalTweaker
[19:35:12] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.TerminalTweaker
[19:35:12] [main/INFO] [net.labymod.support.util.CapturePrintStream]: OptiFineForgeTweaker: getLaunchArguments
[19:35:12] [main/INFO] [LaunchWrapper]: Launching wrapped minecraft {net.minecraft.client.main.Main}
[19:35:13] [main/INFO] [Config]: [OptiFine] *** Reflector Forge ***
[19:35:13] [main/INFO] [Config]: [OptiFine] (Reflector) Class not present: mods.betterfoliage.client.BetterFoliageClient
[19:35:13] [main/INFO] [Config]: [OptiFine] (Reflector) Class not present: net.minecraftforge.fml.common.asm.transformers.BlamingTransformer
[19:35:14] [main/ERROR] [net.labymod.support.util.CapturePrintStream]: Catching
java.lang.ExceptionInInitializerError: null
	at net.minecraftforge.common.ForgeHooks.<clinit>(ForgeHooks.java:339) ~[ForgeHooks.class:?]
	at java.lang.Class.forName0(Native Method) ~[?:1.8.0_51]
	at java.lang.Class.forName(Class.java:264) ~[?:1.8.0_51]
	at net.optifine.reflect.ReflectorClass.getTargetClass(ReflectorClass.java:47) [ReflectorClass.class:?]
	at net.optifine.reflect.ReflectorClass.<init>(ReflectorClass.java:26) [ReflectorClass.class:?]
	at net.optifine.reflect.ReflectorClass.<init>(ReflectorClass.java:17) [ReflectorClass.class:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:264) [Reflector.class:?]
	at net.minecraft.client.renderer.vertex.DefaultVertexFormats.<clinit>(DefaultVertexFormats.java:16) [cdy.class:?]
	at net.minecraftforge.client.ForgeHooksClient$LightGatheringTransformer.<clinit>(ForgeHooksClient.java:602) [ForgeHooksClient$LightGatheringTransformer.class:?]
	at net.minecraftforge.client.ForgeHooksClient.<clinit>(ForgeHooksClient.java:645) [ForgeHooksClient.class:?]
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:335) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
Caused by: java.lang.RuntimeException: Accessed Items before Bootstrap!
	at net.minecraft.init.Items.<clinit>(SourceFile:9) ~[air.class:?]
	... 18 more
[19:35:14] [main/INFO] [net.labymod.support.util.CapturePrintStream]: Caused by: java.lang.RuntimeException: Accessed Items before Bootstrap!
[19:35:14] [main/WARN] [Config]: [OptiFine] (Reflector) More than one method found: net.minecraft.potion.Potion.renderHUDEffect
[19:35:14] [main/WARN] [Config]: [OptiFine] (Reflector)  - public void net.minecraft.potion.Potion.renderHUDEffect(int,int,net.minecraft.potion.PotionEffect,net.minecraft.client.Minecraft,float)
[19:35:14] [main/WARN] [Config]: [OptiFine] (Reflector)  - public void net.minecraft.potion.Potion.renderHUDEffect(net.minecraft.potion.PotionEffect,net.minecraft.client.gui.Gui,int,int,float,float)
[19:35:14] [main/INFO] [Config]: [OptiFine] *** Reflector Vanilla ***
[19:35:14] [Client thread/INFO] [net.minecraft.client.Minecraft]: Setting user: MrFreakzz
[19:35:16] [Client thread/ERROR] [LaunchWrapper]: Unable to launch
java.lang.reflect.InvocationTargetException: null
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.minecraft.init.Items
	at net.minecraft.potion.PotionHelper.func_185207_a(PotionHelper.java:154) ~[akh.class:?]
	at net.minecraft.init.Bootstrap.func_151354_b(Bootstrap.java:504) ~[ni.class:?]
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:365) ~[bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) ~[Main.class:?]
	... 6 more
[19:35:16] [Client thread/ERROR] [net.labymod.support.util.CapturePrintStream]: Catching
net.minecraftforge.fml.relauncher.FMLSecurityManager$ExitTrappedException: null
	at net.minecraftforge.fml.relauncher.FMLSecurityManager.checkPermission(FMLSecurityManager.java:49) ~[forge-1.12.2-14.23.5.2847.jar:?]
	at java.lang.SecurityManager.checkExit(SecurityManager.java:761) ~[?:1.8.0_51]
	at java.lang.Runtime.exit(Runtime.java:107) ~[?:1.8.0_51]
	at java.lang.System.exit(System.java:971) ~[?:1.8.0_51]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:138) ~[launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) ~[launchwrapper-1.12.jar:?]
`

**Comment by realjackasterisk** [comment]
Please post the log on pastebin, it's completely unreadable here. What mods are you using?

**Comment by FreakzzYT** [comment]
Thanks for replying! My current mods are: _BetterFPS 1.4.6, LabyMod, LunatriusCore 1.2.0.42 universal, Schematica 1.8.0.169 universal_
I also noticed something else, when I don't have the optifine mod in my mods folders, my client freezes for a split second and resumes. It keeps on happening and I don't know how to stop it.
Regarding the incident with my client crashing, whenever I place Optifine 1.12.2 HD U E2 and press play, it loads, closes the launcher only to reopen it a few seconds later telling me it crashed. 
I've listed both situations on pastebin:
[With optifine](https://pastebin.com/F9K8gBmp)
[Without optifine](https://pastebin.com/AWVuWTYS)
